ext/opcache/tests/gh8846.phpt

```diff
@@ -0,0 +1,59 @@
+--TEST--
+Bug GH-8846: Delayed early binding can be used for classes without parents
+--EXTENSIONS--
+opcache
+--CONFLICTS--
+server
+--INI--
+opcache.validate_timestamps=1
+opcache.revalidate_freq=0
+--FILE--
+<?php
+
+file_put_contents(__DIR__ . '/gh8846-index.php', <<<'PHP'
+<?php
+if (!@$_GET['skip']) {
+    include __DIR__ . '/gh8846-1.php';
+}
+include __DIR__ . '/gh8846-2.php';
+echo "Ok\n";
+PHP);
+
+file_put_contents(__DIR__ . '/gh8846-1.php', <<<'PHP'
+<?php
+class Foo {
+	const BAR = true;
+}
+PHP);
+
+file_put_contents(__DIR__ . '/gh8846-2.php', <<<'PHP'
+<?php
+var_dump(Foo::BAR);
+class Foo {
+	const BAR = true;
+}
+PHP);
+
+// FIXME: Avoid this sleep
+sleep(2);
```

Comment:
That didn't work, but I was able to avoid the issue by using an `*.inc` file just for the included files (the built-in server doesn't work with anything but `*.php` files).